app/controllers/search_controller.rb

```diff
@@ -38,7 +38,7 @@ def pattern_search
       return
     else
       flash_error(:runtime_invalid.t(type: :search, value: type.inspect))
-      redirect_back_or_default(:root)
+      redirect_back_or_default(observations_path)
```

Comment:
The point of re-directing to `:root` in `bootstrap-4` was to change behavior for those not logged in. I'm not sure if this is intentional, but  this looks like a change in behavior from `master` for logged-in users. The corresponding `master` line is:
```ruby
redirect_back_or_default(action: :list_rss_logs)
```
(FWIW, I think the `master` behavior is right (at least for logged-in users). If I somehow mess up the search type, I want to go back to my home page.)

Reply:
Got it. Reverting and re-testing!

Reply:
Yes, this is definitely the whole idea behind dual-root .Logged in users in bootstrap-4 should be returned to their familiar Activity Log; not-logged-in users should never be redirected there, unless they navigate to it expressly. 

The working logic is: only people with some investment in the site are likely to understand what the Activity Log is recording, to have any interest in it, or to have any business with it. Casual visitors can safely be assumed to be on MO almost exclusively to browse observations, or maybe names, although they can of course browse wherever they like.